internal/langserver/handlers/did_change_watched_files.go

```diff
@@ -0,0 +1,79 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	"github.com/creachadair/jrpc2"
+	"github.com/hashicorp/terraform-ls/internal/job"
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	"github.com/hashicorp/terraform-ls/internal/protocol"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+	"github.com/hashicorp/terraform-ls/internal/terraform/ast"
+)
+
+func (svc *service) DidChangeWatchedFiles(ctx context.Context, params lsp.DidChangeWatchedFilesParams) error {
+	var ids job.IDs
+
+	for _, change := range params.Changes {
+		uri := string(change.URI)
+		fi, err := os.Stat(uri)
+		if err != nil {
+			jrpc2.ServerFromContext(ctx).Notify(ctx, "window/showMessage", &lsp.ShowMessageParams{
+				Type:    lsp.Warning,
+				Message: fmt.Sprintf("Unable to update: %s, Failed to open directory", uri),
+			})
+			continue
+		}
+
+		if !fi.IsDir() {
```

Comment:
Ok, so I pushed this change but I'm a little unsure of what's going on here. Can you corroborate the following? We're not handling single file changes here, although the spec says it can be part of the request, and we [_have_](https://github.com/hashicorp/terraform-ls/pull/843) preliminary support for the same (#843), and so we return with a warning if it's not a directory?

Reply:
I think the linked single file support won't affect the events we receive here. You are right however that we may receive file changes. IMO the best course of action would be simply using that path straight away as a module path, i.e. avoid calling `path.Base()`

Reply:
Got it, also doing away with the `IsDir()` check here now.